### PR TITLE
Spatial query

### PR DESCRIPTION
**Pull request by qgan7125**

### CODE REVIEW

File: src/components/current-query/index.js [hunk 1]
```diff
@@ -59,8 +69,9 @@ class CurrentQuery extends React.Component {
   render() {
     const {bootstrapCss, query} = this.props;
 
+
     const splitFields = query.searchFields
-      .filter((searchField) => searchField.value && searchField.value.length > 0)
+      .filter((searchField) => searchField.value && (searchField.value.length > 0 || Object.keys(searchField.value).length > 0))
```
Comment (qgan7125) [inline src/components/current-query/index.js:74]: The value of spatial query type is the dictionary. We need to check if the dictionary has value.

File: src/components/spatial-query/index.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
+import PropTypes from 'prop-types';
+import React from "react";
+import cx from "classnames";
+
+class SpatialQuery extends React.Component {
+  constructor(props) {
+    super(props);
+
```
Comment (dannymandel) [inline src/components/spatial-query/index.js:8]: Can we store these values with numbers instead of strings?

Comment (dannymandel) [reply src/components/spatial-query/index.js:8]: It seems strange to write strings in there when we actually want to represent floating point numbers

Comment (dannymandel) [reply src/components/spatial-query/index.js:8]: Or maybe you're using them as placeholders to indicate empty?  I'm not quite sure I understand.

Comment (qgan7125) [reply src/components/spatial-query/index.js:8]: Yeah, these values are just the placeholders to indicate empty.

Comment (qgan7125) [reply src/components/spatial-query/index.js:8]: I don't think we should use 0 or some numbers to represent the empty values.

Comment (dannymandel) [reply src/components/spatial-query/index.js:8]: OK, I think that makes sense.  Maybe just add a comment indicating that's why we're doing it and write that the actual values that get stored are numbers.

**Comment by dannymandel** [comment]
This looks good to me, please add a comment about how the default values are being used and why they are being used that way.

**Review by dannymandel — APPROVED** [review]

File: src/components/spatial-query/index.js (hunk 2 above)
Comment (qgan7125) [reply src/components/spatial-query/index.js:8]: Ok

**Pull request merged**